test/messagesIncoming.ts

```diff
@@ -252,42 +252,98 @@ describe('Incoming Messages', async () => {
-			expect(await env.fuelMessagePortal.owner()).to.not.be.equal(signer1);
+		it('Should be able to grant admin role', async () => {
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer1)).to.equal(false);
 
-			// Transfer ownership
-			await expect(env.fuelMessagePortal.transferOwnership(signer1)).to.not.be.reverted;
-			expect(await env.fuelMessagePortal.owner()).to.be.equal(signer1);
+			// Grant admin role
+			await expect(env.fuelMessagePortal.grantRole(defaultAdminRole, signer1)).to.not.be.reverted;
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer1)).to.equal(true);
 		});
 
-		it('Should not be able to switch owner as non-owner', async () => {
-			expect(await env.fuelMessagePortal.owner()).to.be.equal(signer1);
+		it('Should be able to renounce admin role', async () => {
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer0)).to.equal(true);
 
-			// Attempt transfer ownership
-			await expect(env.fuelMessagePortal.transferOwnership(signer0)).to.be.revertedWith(
-				'Ownable: caller is not the owner'
+			// Revoke admin role
+			await expect(env.fuelMessagePortal.renounceRole(defaultAdminRole, signer0)).to.not.be.reverted;
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer0)).to.equal(false);
+		});
+
+		it('Should not be able to grant admin role as non-admin', async () => {
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer0)).to.equal(false);
+
+			// Attempt grant admin role
+			await expect(env.fuelMessagePortal.grantRole(defaultAdminRole, signer0)).to.be.revertedWith(
+				`AccessControl: account ${env.addresses[0].toLowerCase()} is missing role ${defaultAdminRole}`
+			);
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer0)).to.equal(false);
+		});
+
+		it('Should be able to grant then revoke admin role', async () => {
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer0)).to.equal(false);
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer1)).to.equal(true);
+
+			// Grant admin role
+			await expect(env.fuelMessagePortal.connect(env.signers[1]).grantRole(defaultAdminRole, signer0)).to.not.be
+				.reverted;
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer0)).to.equal(true);
+
+			// Revoke previous admin
+			await expect(env.fuelMessagePortal.revokeRole(defaultAdminRole, signer1)).to.not.be.reverted;
+			expect(await env.fuelMessagePortal.hasRole(defaultAdminRole, signer1)).to.equal(false);
+		});
+
+		it('Should be able to grant pauser role', async () => {
+			expect(await env.fuelMessagePortal.hasRole(pauserRole, signer1)).to.equal(false);
+
+			// Grant pauser role
+			await expect(env.fuelMessagePortal.grantRole(pauserRole, signer1)).to.not.be.reverted;
+			expect(await env.fuelMessagePortal.hasRole(pauserRole, signer1)).to.equal(true);
+		});
+
+		it('Should be able to grant permission as pauser', async () => {
```

Comment:
This says it should grant but the test makes sure that nothing is granted. Should there be a `not` in there somewhere e.g. `Should not ... `

Reply:
Yeah, it should say `Should not be able to grant permission as pauser`. Thanks!